File: app/src/main/java/com/ebsoft/watchlist/data/model/IEX/Quote.java

package com.ebsoft.watchlist.data.model.IEX;
import com.ebsoft.watchlist.data.model.StockInfo;
import com.ebsoft.watchlist.utils.Constants;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Quote implements StockInfo {

    @SerializedName("symbol")
    @Expose
    private String symbol;

    @SerializedName("latestPrice")
    @Expose
    private Double latestPrice;

    @SerializedName("change")
    @Expose
    private Double change;

    @SerializedName("changePercent")
    @Expose
    private Double changePercent;


    @Override
    public String getSymbol() {
        return symbol;
    }

    @Override
    public float getPrice() {
        return latestPrice.floatValue();
    }

    @Override
    public float getChange() {
        return change.floatValue() * Constants.IEX_API_PERCENTAGE_MULTIPLIER;
    }

    @Override
    public float getChangePercent() {
        return changePercent.floatValue() * Constants.IEX_API_PERCENTAGE_MULTIPLIER;
    }
}

File: app/src/main/java/com/ebsoft/watchlist/data/model/IEX/Quote.kt
package com.ebsoft.watchlist.data.model.IEX

import com.ebsoft.watchlist.data.model.StockInfo
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class Quote (

    @SerializedName("symbol")
    @Expose
    override val symbol: String = "",

    @SerializedName("latestPrice")
    @Expose
    override val price: Float = 0.0f,

    @SerializedName("change")
    @Expose
    override val change: Float = 0.0f,

    @SerializedName("changePercent")
    @Expose
    override val changePercent: Float = 0.0f) : StockInfo